Code/Tools/AssetProcessor/native/unittests/RCcontrollerUnitTests.cpp

```diff
@@ -214,315 +158,455 @@ void RCcontrollerUnitTests::RunRCControllerTests()
     tempJobNames << "c:/somerandomfolder/pqr/123.456";
 
     // test - compile group for an exact ID succeeds when that exact ID is called.
-
-    QList<RCJob*> createdJobs;
-
-
-
     for (QString name : tempJobNames)
     {
         AZ::Uuid uuidOfSource = AZ::Uuid::CreateName(name.toUtf8().constData());
-        RCJob* job = new RCJob(rcJobListModel);
+        RCJob* job = new RCJob(m_rcJobListModel);
         AssetProcessor::JobDetails jobDetails;
         jobDetails.m_jobEntry.m_watchFolderPath = "c:/somerandomfolder/dev";
         jobDetails.m_jobEntry.m_databaseSourceName = jobDetails.m_jobEntry.m_pathRelativeToWatchFolder = name;
         jobDetails.m_jobEntry.m_platformInfo = { "pc",{ "desktop", "renderer" } };
         jobDetails.m_jobEntry.m_jobKey = "Compile Stuff";
         jobDetails.m_jobEntry.m_sourceFileUUID = uuidOfSource;
         job->Init(jobDetails);
-        rcJobListModel->addNewJob(job);
-        createdJobs.push_back(job);
+        m_rcJobListModel->addNewJob(job);
+        m_createdJobs.push_back(job);
     }
 
     // double them up for "android" to make sure that platform is respected
     for (QString name : tempJobNames)
     {
         AZ::Uuid uuidOfSource = AZ::Uuid::CreateName(name.toUtf8().constData());
-        RCJob* job0 = new RCJob(rcJobListModel);
+        RCJob* job0 = new RCJob(m_rcJobListModel);
         AssetProcessor::JobDetails jobDetails;
         jobDetails.m_jobEntry.m_databaseSourceName = jobDetails.m_jobEntry.m_pathRelativeToWatchFolder = name;
         jobDetails.m_jobEntry.m_platformInfo = { "android" ,{ "mobile", "renderer" } };
         jobDetails.m_jobEntry.m_jobKey = "Compile Other Stuff";
         jobDetails.m_jobEntry.m_sourceFileUUID = uuidOfSource;
         job0->Init(jobDetails);
-        rcJobListModel->addNewJob(job0);
+        m_rcJobListModel->addNewJob(job0);
     }
 
-    bool gotCreated = false;
-    bool gotCompleted = false;
-    NetworkRequestID gotGroupID;
-    AssetStatus gotStatus = AssetStatus_Unknown;
-    QObject::connect(&m_rcController, &RCController::CompileGroupCreated, this, [&](NetworkRequestID groupID, AssetStatus status)
+    ConnectCompileGroupSignalsAndSlots(gotCreated, gotCompleted, gotGroupID, gotStatus);
+}
+
+void RCcontrollerUnitTests::Reset()
+{
+    m_rcController->m_RCJobListModel.m_jobs.clear();
+    m_rcController->m_RCJobListModel.m_jobs.clear();
```

Comment:
Duplicate line

Also does this actually called delete on the jobs in the queue?